framework/src/base/EigenProblem.C

```diff
@@ -190,3 +190,15 @@ EigenProblem::computeResidualType(const NumericVector<Number> & soln,
       _is_residual_initialed = true;
   }
 }
+
+void
+EigenProblem::computeResidualABType(const NumericVector<Number> & soln,
+                                    NumericVector<Number> & Ax,
+                                    NumericVector<Number> & Bx,
+                                    Moose::KernelType type)
+{
+  FEProblemBase::computeResidualType(soln, Ax, type);
```

Comment:
now passing `Ax` in this function does not make sense because of tagging. Instead, we should have a new function to associate a vector to a tag like `setTaggedVector(TagName tag, NumericVector<Number> & x)`. I am also curious who should be creating the vector, PETSc or MOOSE?

Reply:
Good question. If I let PETSc create vectors, I would worry about MOOSE manipulate the vectors in arbitrary way. This potentially be very buggy.  I

I would like to have MOOSE create their own vectors, and PETSc also has their own vectors. Once the MOOSE vectors are filled, they will be copied over to the PETSc vectors. In this way, I won't worry about MOOSE changes things inside of PETSc. The disadvantage is that we hold extra copies. The number of tagged vectors are supposed to be small.

Reply:
Any suggestions are welcome on this.